plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/internal/edit/commands/StraightenToCommand.java

```diff
@@ -147,14 +147,18 @@ public StraightenToCommand(AbstractDiagramEdgeEditPart edgeEditPart, int straigh
         super(edgeEditPart.getEditingDomain(), StraightenToAction.getLabel(straightenType), null);
         this.straightenType = straightenType;
         for (AbstractDiagramEdgeEditPart selectedEdgeEditPart : selectedEdgeEditParts) {
-            StraightenToCommandData straightenToCommandData = new StraightenToCommandData();
-            if (selectedEdgeEditPart.getSource() instanceof IGraphicalEditPart) {
-                straightenToCommandData.sourceEditPart = (IGraphicalEditPart) selectedEdgeEditPart.getSource();
-            }
-            if (selectedEdgeEditPart.getTarget() instanceof IGraphicalEditPart) {
-                straightenToCommandData.targetEditPart = (IGraphicalEditPart) selectedEdgeEditPart.getTarget();
+            if (!(selectedEdgeEditPart.getSource() == null && selectedEdgeEditPart.getTarget() == null)) {
```

Comment:
While ignore only the case where both source *and* target are null?
Is there a reason not to ignore the cases where only one of source *or* target is null?

Reply:
Later, there is a `if (data.moveSource) {` (line 527), so we can imagine that the command is "OK" only with source or target filled. This case has been seen only in automatic test, so I prefer to handle only the observed case.